flink-connectors/flink-sql-connector-kinesis/pom.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<artifactId>flink-connectors</artifactId>
+		<groupId>org.apache.flink</groupId>
+		<version>1.12-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-sql-connector-kinesis_${scala.binary.version}</artifactId>
+	<name>Flink : Connectors : SQL : Kinesis</name>
+
+	<packaging>jar</packaging>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-kinesis_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>shade-flink</id>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<shadeTestJar>false</shadeTestJar>
+							<artifactSet>
+								<includes>
+									<include>*:*</include>
+								</includes>
+								<excludes>
+									<exclude>org.apache.flink:force-shading</exclude>
+								</excludes>
+							</artifactSet>
+							<filters>
+								<filter>
+									<artifact>*:*</artifact>
+									<excludes>
+										<exclude>codegen-resources/**</exclude>
```

Comment:
These 4 excludes are bundled into the `flink-kinesis-connector` and result in the shading E2E check failing. I have removed them from here instead of the source connector because I am not 100% sure if they are used at runtime. For instance, it looks like the `mime.types` is used by S3, but not sure if this is actually used by connector. I think it is safer to leave them in the connector and remove from the SQL bundle, without performing a more thorough investigation. Thoughts?

Reply:
If they are used at runtime, I assume that you will 

* avoid runtime errors when depending on `flink-connector-kinesis`.
* hit runtime errors when depending on `flink-sql-connector-kinesis` at runtime.

Is there a quick way to make the test pass without excluding those? If not, I maybe we can  remove them from both and re-include if people start complaining (seems more consistent).

Reply:
> Is there a quick way to make the test pass without excluding those? 
Yes

> Maybe we can remove them from both and re-include if people start complaining
Or this

Either option is a quick change, lets see what @twalthr 's preference is

Reply:
I would keep them in `flink-connector-kinesis` and fully rely on Maven + manual dependency management there. For the SQL jars we do best effort dependency management to help less skilled users. But they always have the fallback to go to the regular `flink-connector-kinesis`.